setup/setup.sh

```diff
@@ -1,62 +1,100 @@
+
+    cp $USERNAME.key $MINIKUBE_DIR/$USERNAME.key
+    cp $USERNAME.crt $MINIKUBE_DIR/$USERNAME.crt
+}
+
+delete_certificates() {
+    echo "Deleting certificates..."
+    if ! rm ${MINIKUBE_DIR}/$USERNAME.key ; then
+        echo "Error deleting key file ${MINIKUBE_DIR}/$USERNAME.key"
+        exit
+    fi
+
+    if ! rm ${MINIKUBE_DIR}/$USERNAME.crt; then
+        echo "Error deleting certificate file ${MINIKUBE_DIR}/$USERNAME.crt"
+        exit
+    fi
+}
+
+create_namespace() {
+    echo "Creating namespace '${1}'..."
+    if ! kubectl get namespace ${1}  > /dev/null 2>&1 ; then
+        if ! kubectl create namespace ${1}  > /dev/null 2>&1 ; then
+            echo "Error while creating namespace ${1}"
+            exit
+        fi
+    fi
 }
 
-function create_namespace() {
-    echo "Creating namespace 'redhat-test'..."
-    if ! kubectl get namespace redhat-test &> /dev/null ; then
-        if ! kubectl create namespace redhat-test &> /dev/null ; then
-            echo "Error while creating namespace"
+delete_namespace() {
+    echo "Deleting namespace '${1}'..."
+    if kubectl get namespace ${1}  > /dev/null 2>&1 ; then
+        if ! kubectl delete namespace ${1}  > /dev/null 2>&1 ; then
+            echo "Error while deleting namespace ${1}"
             exit
         fi
     fi
 }
 
-function configure_kubectl_credentials() {
-    echo "Creating Kubectl credentials and context..."
-    if ! kubectl config set-credentials $USERNAME --client-certificate=$USERNAME.crt --client-key=$USERNAME.key &> /dev/null ; then
+configure_kubectl_credentials() {
+    echo "Creating Kubectl credentials ..."
+    if ! kubectl config set-credentials $USERNAME --client-certificate=$USERNAME.crt --client-key=$USERNAME.key  > /dev/null 2>&1 ; then
         echo "Error while creating config credentials"
         exit
     fi
+}
 
-    if ! kubectl config set-context $USERNAME-context --cluster=minikube --user=$USERNAME --namespace=redhat-test &> /dev/null; then
+create_kubectl_context() {
+    echo "Creating Kubectl context $MINIKUBE_CONTEXT for namespace ${1} ..."
+    if ! kubectl config set-context $MINIKUBE_CONTEXT --cluster=minikube --user=$USERNAME --namespace=${1}  > /dev/null 2>&1; then
         echo "Error while creating config context"
         exit
     fi
 }
 
-function apply_role_resources() {
-    if ! kubectl apply -f $SCRIPTPATH/files/role.yml &> /dev/null ; then
-        echo "Could not apply Role resource"
+delete_kubectl_context() {
+    echo "Deleting Kubectl context ${MINIKUBE_CONTEXT} ..."
+    if ! kubectl config delete-context $MINIKUBE_CONTEXT  > /dev/null 2>&1; then
+        echo "Error while deleting config context"
         exit
     fi
+}
 
-    if ! kubectl apply -f $SCRIPTPATH/files/role-binding.yml &> /dev/null ; then
+apply_role_resources() {
+    if ! sed "s/{username}/${USERNAME}/g; s/{namespace}/${NAMESPACE_DEV}/g" $SCRIPTPATH/files/role-binding.yml | kubectl apply -f -  > /dev/null 2>&1 ; then
```

Comment:
We need to use the parameter of the function
```suggestion
    if ! sed "s/{username}/${USERNAME}/g; s/{namespace}/${1}/g" $SCRIPTPATH/files/role-binding.yml | kubectl apply -f -  > /dev/null 2>&1 ; then
```

Reply:
Good catch!